agent.go

```diff
@@ -193,9 +193,12 @@ func (c *controller) agentSetup() error {
 		}
 	}
 
+	c.Lock()
 	if c.agent != nil {
 		close(c.agentInitDone)
+		c.agentInitDone = make(chan struct{})
```

Comment:
agentSetup() and SetupIngress (caller of AgentInitWait) can get called from different async paths. With the earlier behavior of only closing the channel, even if SetupIngress gets called later it would return immediately if the agent has already been setup. But by creating the channel right after closing it won't it result in AgentInitWait waiting forever if SetupIngress gets called later ?